From 44eecefc62b87a5495e712f1eaad9d32e834b9fa Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 21 Jan 2021 14:21:32 +0200
Subject: [PATCH] [SW-29389] gaudi: security: protect psoc etr fix

It appears we should not protect the whole ETR page, as it will be
configured by the profiler (through LKD) later during its use.
Narrow down the protection to the single register AXICTL which is
responsible for setting the ETR protection level.

Change-Id: I9e827375f36ecc3ce4d11202888d3ce3a0624712
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 874bb63..4928341 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -27,6 +27,7 @@
 #include <sif_rtr_ctrl_0_regs.h>
 #include <sif_rtr_ctrl_7_regs.h>
 #include <psoc_global_conf_regs.h>
+#include <psoc_etr_regs.h>
 #include <dma_if_w_s_down_ch0_regs.h>
 #include <dma_if_w_s_regs.h>
 #include <sync_mngr_w_s_regs.h>
@@ -1233,9 +1234,9 @@
 	max = mmNIC4_MAC_CH0_BASE;
 	preboot_secure_lbw_rr(14, min, max);
 
-	/* PSOC_ETR up to PSOC_ETF: RR_15 */
-	min = mmPSOC_ETR_BASE - 1;
-	max = mmPSOC_ETF_BASE;
+	/* mmPSOC_ETR_AXICTL (single register): RR_15 */
+	min = HL_CFG_BASE + mmPSOC_ETR_AXICTL - 1;
+	max = HL_CFG_BASE + mmPSOC_ETR_AXICTL + sizeof(u32);
 	preboot_secure_lbw_rr(15, min, max);
 }
 
